docs/en/design/logging.md

```diff
@@ -1,3 +1,448 @@
+history_buffer = HistoryBuffer([1, 2, 3], [1, 1, 1])
+history_buffer.update(4)  # count default to 1
+log_history, count_history = history_buffer.data
+# [1, 2, 3, 4] [1, 1, 1, 1]
+history_buffer.update(5, 2)
+log_history, count_history = history_buffer.data
+# [1, 2, 3, 4, 5] [1, 1, 1, 1, 2]
+```
+
+### Basic statistical methods
+
+HistoryBuffer provides some basic statistical methods:
+
+- `current()`: Get the latest data.
+- `mean(window_size=None)`: Count the mean value of the previous `window_size` data. Defaults to None, as global mean.
+- `max(window_size=None)`: Count the max value of the previous `window_size` data. Defaults to None, as global maximum.
+- `min(window_size=None)`: Count the min value of the previous `window_size` data. Defaults to None, as global minimum.
+
+```python
+history_buffer = HistoryBuffer([1, 2, 3], [1, 1, 1])
+history_buffer.min(2)
+# 2, the minimum in [2, 3]
+history_buffer.min()
+# 1, the global minimum
+
+history_buffer.max(2)
+# 3，the maximum in [2, 3]
+history_buffer.min()
+# 3, the global maximum
+history_buffer.mean(2)
+# 2.5，the mean value in [2, 3], (2 + 3) / (1 + 1)
+history_buffer.mean()
+# 2, the global mean, (1 + 2 + 3) / (1 + 1 + 1)
+history_buffer = HistoryBuffer([1, 2, 3], [2, 2, 2])  # Cases when counts are not 1
+history_buffer.mean()
+# 1, (1 + 2 + 3) / (2 + 2 + 2)
+history_buffer = HistoryBuffer([1, 2, 3], [1, 1, 1])
+history_buffer.update(4, 1)
+history_buffer.current()
+# 4
+```
+
+### Register statistical methods
+
+In order to make HistoryBuffer extensible, we provide `register_statistics` method for users to register their custom statistical methods.
+
+```python
+from mmengine.logging import HistoryBuffer
+import numpy as np
+
+
+@HistoryBuffer.register_statistics
+def weighted_mean(self, window_size, weight):
+    assert len(weight) == window_size
+    return (self._log_history[-window_size:] * np.array(weight)).sum() / \
+            self._count_history[-window_size:]
+
+
+history_buffer = HistoryBuffer([1, 2], [1, 1])
+history_buffer.statistics('weighted_mean', 2, [2, 1])  # get (2 * 1 + 1 * 2) / (1 + 1)
+```
+
+You can call `statistics` to use registered statistical method by providing method name and extra arguments, as explained below.
+
+### Entry point of statistical methods
+
+We provide a method, signatured as `statistics(name, *args, **kwargs)`, in HistoryBuffer to call any statistical method with any number of arguments for extensibility. The `name` parameter should be a registered method name (i.e. built-in methods like `min` and `max`), while `*args` and `**kwargs` should be corresponding method's arguments.
+
+```python
+history_buffer = HistoryBuffer([1, 2, 3], [1, 1, 1])
+history_buffer.statistics('mean')
+# 2, as global mean
+history_buffer.statistics('mean', 2)
+# 2.5, as the mean of [2, 3]
+history_buffer.statistics('mean', 2, 3)
+# Error! mismatch arguments given to `mean(window_size)`
+history_buffer.statistics('data')
+# Error! `data` method not registered
+```
```

Comment:
```suggestion
### Statistical Methods Invoking

Statistical methods can be accessed through `HistoryBuffer.statistics` with method name and arguments. The `name` parameter should be a registered method name (i.e. built-in methods like `min` and `max`), while arguments should be the corresponding method's arguments.

\```python
history_buffer = HistoryBuffer([1, 2, 3], [1, 1, 1])
history_buffer.statistics('mean')
# 2, as global mean
history_buffer.statistics('mean', 2)
# 2.5, as the mean of [2, 3]
history_buffer.statistics('mean', 2, 3)
# Error! mismatch arguments given to `mean(window_size)`
history_buffer.statistics('data')
# Error! `data` method not registered
\```

### Statistical Methods Registration

Custom statistical methods can be registered through `@HistoryBuffer.register_statistics`.

\```python
from mmengine.logging import HistoryBuffer
import numpy as np


@HistoryBuffer.register_statistics
def weighted_mean(self, window_size, weight):
    assert len(weight) == window_size
    return (self._log_history[-window_size:] * np.array(weight)).sum() / \
            self._count_history[-window_size:]


history_buffer = HistoryBuffer([1, 2], [1, 1])
history_buffer.statistics('weighted_mean', 2, [2, 1])  # get (2 * 1 + 1 * 2) / (1 + 1)
\```

```

Reply:
Mark, Chinese docs should also be changed